pkg/cvo/internal/generic.go

```diff
@@ -46,17 +49,37 @@ func applyUnstructured(ctx context.Context, client dynamic.ResourceInterface, re
 		return nil, false, nil
 	}
 
-	existing.SetAnnotations(required.GetAnnotations())
-	existing.SetLabels(required.GetLabels())
-	existing.SetOwnerReferences(required.GetOwnerReferences())
 	skipKeys := sets.NewString("apiVersion", "kind", "metadata", "status")
+
+	// create a copy of required, but copy skipKeys from existing
+	// this would copy skipKeys data into expected from existing
+	expected := required.DeepCopy()
+	for k, v := range existing.Object {
+		if skipKeys.Has(k) {
+			expected.Object[k] = v
```

Comment:
This sort of thing is why I think generically logging diffs is brittle.  I'd rather have the `Ensure*` helpers return a slice of patches instead of the current `modified` boolean, so we could log the patches "Updating $PROPERTY to $VALUE to clobber $EXISTING_VALUE" instead of having logic about the things we care about clobbering both here and in the `Ensure*` helpers.

Reply:
Logging everytime `modified` is changed may be too verbose. I think we should do that later, on level 4 so that we wouldn't hit master's disks with too much log lines.

Reply:
if, instead of twiddling `*modified`, these helpers return `[]patch, error`, we can log any `patch` we get at one time, before calling the `UPDATE`.  Should be same number of log lines, and we'd only complain about the in-cluster properties we were trying to stomp.

Reply:
This is only required if we want to see not only that we're hot looping but also why. The latter could be dropped and once we know we're hot looping figuring out the why becomes part of the fix process done off line. Choosing one over the other is a question of how often we believe this is going to occur. If rarely then we probably don't need the "why" in the baseline.

Reply:
CVO can start hotlooping on any non-standard configuration - and its complicated to find out details to fix it. 
I'd prefer to include "why" in every run, so that we wouldn't have to request customers to reconfigure CVO to get details.

Reply:
@vrutkovs What sort of non-standard configs would they have which are managed by the CVO that we'd want to look into?

Reply:
Users can [sideload additional manifests](https://github.com/openshift/installer/blob/master/docs/user/customization.md#control-plane-with-no-taints) during bootstrap